### ISSUE TITLE
Improve Versioning from a User Perspective

### ISSUE THREAD

**Issue by Crain-32** [ssrando/ssrando#551]
After spending 30 minutes dealing with version mismatch in a casual race yesterday, here are some general improvements to the versioning that could solve simple problems for both technical/non-technical users.

1. Embed the Version in the Setting String.
Setting Strings working between versions is great, however if I fail to check with my friends what version of the randomizer they're running, we won't know that we're dealing with a version mismatch till we randomize. By embedding the generated version in the setting string, when I paste it into my randomizer I could easily be informed "this is from a different version". 

2. Improve Version Discrimination 
It is very difficult looking at multiple versions of the randomizer to know which one is "latest". For example I currently have around 3 different 2.2.0_HASH versions of the randomizer on my computer, and hashes are not the fastest thing to share.
Releases should be easy for users to tell them apart, and developers shouldn't have to consult the commit tree to know where two versions are in relationship to each other. I'd recommend the following adjustment to the naming.

Latest Development build,
[RANDOMIZER_VERSION]-DEV.<GIT_WORKFLOW_COUNT>-<SHORTEN_HASH>
For example, the artifact produced after merging a PR with 2.2.0 as the latest stable, using workflow 34 would be
2.2.0-DEV.34-4fa3232
The next PR would be merged, start workflow 35, and produce the following.
2.2.0-DEV.35-3423faa

For fast debugging I know the hash it was built on, and I can identify which version was built before/after another, and it clearly is not a stable build. 

Official releases can remain on the current versioning system.

To summarize, the goal of these changes is to achieve the following.
- As I user I shouldn't need to compare a hash to know if I have the same version as someone else.
- If I receive a seed from someone else I should know I have the current randomizer version when I paste it.
- Development Releases and Stable Releases should be easily distinguished from each other, and I should know which one is more recent without pulling up the git tree.
